### PR TITLE
docs: Add compatibility section to debugging page

### PR DESCRIPTION
**Pull request by polarathene**
- Compatibility troubleshooting advice taken from: https://github.com/docker-mailserver/docker-mailserver/pull/3244#discussion_r1162337652
- Preliminary Checks section moves two checklist items from bug report template: https://github.com/docker-mailserver/docker-mailserver/pull/3381#discussion_r1212454225

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -12,18 +12,13 @@ body:
       label: 📝 Preliminary Checks
       description: Please read these carefully.
       options:
-        - label: I checked that all ports are open and not blocked by my ISP / hosting provider.
-          required: true
-        - label: I know that SSL errors are likely the result of a wrong setup on the user side and not caused by DMS itself. I'm confident my setup is correct.
+        - label: The [documentation on debugging](https://docker-mailserver.github.io/docker-mailserver/latest/config/debugging/), tried the proposed steps to debug the problem, but was still unable to resolve the issue.
```
Comment (polarathene) [inline .github/ISSUE_TEMPLATE/bug_report.yml:15]: This technically makes the next item redundant, depends if you still want them to explicitly see and confirm the expectation of searching previous issues (_we still get users that don't_).

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -12,18 +12,13 @@ body:
...
           required: true
         - label: I searched the issue tracker but was unable to find my issue.
           required: true
-        - label: |
-            I have read:  
-              - The [extended documentation in general](https://docker-mailserver.github.io/docker-mailserver/latest/) but found nothing to resolve the issue;
-              - The [documentation on debugging](https://docker-mailserver.github.io/docker-mailserver/latest/config/debugging/), tried the proposed steps to debug the problem, but was still unable to resolve the issue;
-              - This project's [Code of Conduct](https://github.com/docker-mailserver/docker-mailserver/blob/master/CODE_OF_CONDUCT.md) and I agree;
-              - The [documentation about filing a bug report](https://docker-mailserver.github.io/docker-mailserver/latest/contributing/issues-and-pull-requests/#filing-a-bug-report).
+        - label: The [documentation about filing a bug report](https://docker-mailserver.github.io/docker-mailserver/latest/contributing/issues-and-pull-requests/#filing-a-bug-report).
```
Comment (polarathene) [inline .github/ISSUE_TEMPLATE/bug_report.yml:19]: Also unclear how relevant this is for someone already on the bug report template to confirm. The debug page we link to and the template itself mostly makes this redundant as a checklist item?

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -12,18 +12,13 @@ body:
...
+          required: true
+        - label: This project's [Code of Conduct](https://github.com/docker-mailserver/docker-mailserver/blob/master/CODE_OF_CONDUCT.md) and I agree.
```
Comment (polarathene) [inline .github/ISSUE_TEMPLATE/bug_report.yml:21]: This is fine, but I don't know what value it provides. Someone who checks it and goes against the document likely didn't read it, some interaction happens where we cite it as justification for some action due to them checking that they agree (_not sure if that's actually happened_) :man_shrugging:

File: docs/content/config/debugging.md [hunk 4]
```diff
@@ -10,7 +10,10 @@ This page contains valuable information when it comes to resolving issues you en
 
     Please consider contributing solutions to the [FAQ][docs-faq] :heart:
 
-## Preliminary Information
+## Preliminary Checks
+
+- Check that all published DMS ports are actually open and not blocked by your ISP / hosting provider.
+- SSL errors are likely the result of a wrong setup on the user side and not caused by DMS itself.
```
Comment (polarathene) [inline docs/content/config/debugging.md:16]: If either of these do present themselves as issues, either ignored here or legitimately required more info to troubleshoot, they could be expanded like the below section on outbound port 25 :+1:

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (georglauterbach) [reply .github/ISSUE_TEMPLATE/bug_report.yml:19]: I think this can be removed.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 3 above)
Comment (georglauterbach) [reply .github/ISSUE_TEMPLATE/bug_report.yml:21]: Yeah, can be removed.. or, we may put it like "you automatically agree to our Code of Conduct by submitting an issue".

**Comment by polarathene** [comment]
Feedback applied:
- Moved code of conduct to implicit agreement in description.
- Kept other two items but merged into single check item.

**Review by georglauterbach — APPROVED** [review]

**Pull request merged**